src/components/PupilComponents/QuizResults/QuizResults.tsx

```diff
@@ -0,0 +1,146 @@
+import {
+  OakFlex,
+  OakIcon,
+  OakQuizResultItem,
+  OakSpan,
+} from "@oaknational/oak-components";
+import { isArray } from "lodash";
+
+import { MathJaxProvider } from "@/browser-lib/mathjax/MathJaxProvider";
+import { LessonSectionResults } from "@/components/PupilComponents/LessonEngineProvider";
+import { QuestionsArray } from "@/components/PupilComponents/QuizEngineProvider";
+import { QuizResultQuestionStem } from "@/components/PupilComponents/QuizResultQuestionStem";
+import { QuizResultMCQ } from "@/components/PupilComponents/QuizResultMCQ/QuizResultMCQ";
+import { QuizResultShortAnswer } from "@/components/PupilComponents/QuizResultShortAnswer";
+import { QuizResultOrder } from "@/components/PupilComponents/QuizResultOrder/QuizResultOrder";
+import { QuizResultMatch } from "@/components/PupilComponents/QuizResultMatch";
+import { MathJaxWrap } from "@/browser-lib/mathjax/MathJaxWrap";
+import type { QuizQuestion } from "@/node-lib/curriculum-api-2023/queries/pupilLesson/pupilLesson.schema";
+
+export type QuizResultsProps = {
+  lessonSection: "exit-quiz" | "starter-quiz";
+  quizArray: QuestionsArray;
+  sectionResults: LessonSectionResults;
+};
+export const QuizResults = (props: QuizResultsProps) => {
+  const { lessonSection, quizArray, sectionResults } = props;
+
+  return (
+    <MathJaxProvider>
+      <OakFlex $flexDirection={"column"} $gap={"space-between-xl"} role="list">
+        {sectionResults[lessonSection]?.questionResults &&
+          sectionResults[lessonSection]?.questionResults?.map(
+            (questionResult, index) => {
+              const quizQuestion = quizArray[index];
```

Comment:
I'd break the inner part of the loop into a subcomponent. It's a bit nasty to read like this.